src/arrayMethodJoin.js

```diff
@@ -5,7 +5,34 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let string = '';
+    let sign = separator;
+
+    if (separator === undefined) {
+      sign = ',';
+    }
+
+    if (separator === null) {
+      sign = 'null';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        string += '' + sign;
+      } else {
+        string += this[i] + sign;
+      }
+    }
+
+    const lastElement = this[this.length - 1];
+
+    if (lastElement === null || lastElement === undefined) {
+      string += '';
+    } else {
+      string += lastElement;
+    }
+
```

Comment:
Let's eliminate this, and make all check in the loop

Reply:
Hi, Stanislav! You're right. Thanks! 
Done)